joomla/40-backend-template
# Simplifying Joomla's parameters

### This is a proof of concept and WIP.

As we all know, Joomla has a shit load of parameters and this is overwhelming for new users.

**What this PR does:**
There is now a new parameter in the Global Configuration what allows you to choose the **Tier** (**simple** or **advanced**), which can also be overridden in the User Settings

In the form's XML file, when defining your fields, we can now add `tier="advanced"`, meaning that the form field will only be rendered if your **Tier** is set to **Advanced**

By default, the **Tier** it set to **Simple** as we should put new users first.
### Testing Instructions
1. Go to the Global Configuration
2. In set the **Tier** to **Advanced**, then Save.
3. Go to edit any article and view the "Options" tab
4. You'll see all parameters displaying as normal
5. Go back to the Global Configuration
6. In set the **Tier** to **Simple**, then Save.
7. Go to edit any article and view the "Options" tab
8. You'll now notice the "Linked Titles" and "Position of Article Info" have disappeared.

The user can override the Global Config and chose their own Tier level. In the User Manager, go to edit your user, then go to the **Basic Settings** tab.
### TODO
- [x] Allow users to individually override the GC value
- [ ] Use language constants
- [ ] Apply the advanced tier on parameters we deem necessary.
## Screenshots:

![config](https://cloud.githubusercontent.com/assets/2019801/19769654/3547ca62-9c54-11e6-8a64-b2d5d9bfad26.png)

![screeny](https://cloud.githubusercontent.com/assets/2019801/19769663/3e704c2c-9c54-11e6-8d87-bed5de20920f.jpg)
